### DISCUSSION TITLE
How to use .limit() inside preload with pagination?

### DISCUSSION THREAD

**Discussion by CodeArtisT75** [adonisjs/core#3241] [Help] [answered]
Hello guys.

How can I load limited relations during preload with paginating?

example code:
```javascript
const posts = Post.preload('tags', tagBuilder => {
  tagBuilder.limit(2);
}).paginate(page, perPage);
```

in the example, I want to load 2 tags for each post but it only returns the first 2 tags in all posts.

**Comment by thetutlage** [comment, marked as answer]
Use the `groupLimit` method. https://docs.adonisjs.com/reference/orm/relations/has-many-through#grouplimit

**Reply by CodeArtisT75** [reply]
thank you so much.

that was helpful! :)
